src/main/kotlin/tech/thatgravyboat/skyblockpv/data/museum/RepoMuseumCategory.kt

```diff
@@ -0,0 +1,43 @@
+package tech.thatgravyboat.skyblockpv.data.museum
+
+import com.mojang.serialization.Codec
+import com.mojang.serialization.codecs.RecordCodecBuilder
+import net.minecraft.core.registries.BuiltInRegistries
+import net.minecraft.resources.ResourceLocation
+import net.minecraft.world.item.ItemStack
+import tech.thatgravyboat.skyblockpv.api.ItemAPI
+
+data class RepoMuseumCategory(
+    val name: String,
+    val location: ResourceLocation,
+    val item: Lazy<ItemStack>,
+    val categories: List<String>,
+    val items: List<String>,
+    val priority: Int,
+)
+
+val MUSEUM_CATEGORY_CODEC = RecordCodecBuilder.create {
+    it.group(
+        Codec.STRING.fieldOf("name").forGetter(RepoMuseumCategory::name),
+        ResourceLocation.CODEC.fieldOf("display").forGetter(RepoMuseumCategory::location),
+        Codec.STRING.listOf().fieldOf("categories").forGetter(RepoMuseumCategory::categories),
+        Codec.STRING.listOf().optionalFieldOf("items", emptyList()).forGetter(RepoMuseumCategory::categories),
+        Codec.INT.optionalFieldOf("priority", 0).forGetter(RepoMuseumCategory::priority),
+    ).apply(it) { name, itemLocation, categories, items, priority ->
+        RepoMuseumCategory(
+            name,
+            itemLocation,
+            lazy {
+                if (itemLocation.namespace.equals("minecraft")) {
```

Comment:
move this to its own codec, that is Codec<Lazy<ItemStack>> and instead of throwing an error it will just return a barrier named what it was, also it should just check if "skyblock" and then defer the vanilla logic for other namespaces